src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,197 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(getDriversForCarQuery)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            DriverDaoImpl driverDao = new DriverDaoImpl();
+            while (resultSet.next()) {
+                Optional<Driver> driver = driverDao.get(resultSet.getLong("driver_id"));
+                driver.ifPresent(drivers::add);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get cars drivers by id " + id, e);
+        }
+        return drivers;
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllQuery = "SELECT * FROM cars WHERE is_deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(getAllQuery)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Long id = resultSet.getObject("id", Long.class);
+                Car car = getCarWithManufacturer(resultSet);
+                car.setDrivers(getDriversForCar(id));
+                cars.add(car);
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of drivers.", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCarQuery = "UPDATE cars "
+                + "SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(updateCarQuery)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in cars table", e);
+        }
+        updateCarsDriversTable(car);
+        return car;
+    }
+
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllCarsByDriverQuery = "SELECT cars_drivers.car_id "
+                + "FROM cars_drivers WHERE cars_drivers.driver_id = ?";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(getAllCarsByDriverQuery)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            CarDaoImpl carDao = new CarDaoImpl();
```

Comment:
1. don't need to create an instance of this class to use its methods
2. you should modify your query to get cars' data without using the get method